Add option to create geometric bodies (spheres, prisms, ellipsoids, etc) and compute their effect. (#4583)
GenericMappingTools/gmt: doc/rst/source/supplements/potential/gmtgravmag3d.rst

.. include:: ../../common_SYN_OPTs.rst_

> **gmt gravmag3d** *xyz_file* |-T|\ **v**\ *vert_file* OR |-T|\ **r\|s**\ *raw_file* OR |-M|\ **+**\ *body,params*

Reviewer: ```suggestion
**gmt gravmag3d** *xyz_file* |-T|\ **v**\ *vert_file* OR |-T|\ **r\|s**\ *raw_file* OR |-M|\ **+s**\ *body,params*
```

**-G**\ *outgrid*
    Output the gravity or magnetic anomaly at nodes of this grid file.

.. _-M:

> **-M+s**\ *body,params* (Alaternative to **-Tr**\ /**-Ts**). Create geometric bodies and compute their grav/mag effect.

Reviewer: ```suggestion
**-M+s**\ *body,params* (An alternative to **-Tr**\ /**-Ts**). Create geometric bodies and compute their grav/mag effect.
```

    Select among one or more of the following bodies, where *x0* & *y0* represent the horizontal coordinates
    of the body center [default to 0,0 positive up], *npts* is the number of points that a circle is discretized
    and *n_slices* apply when bodies are made by a pile of slices. For example Spheres and Ellipsoids are made of
    *2 x n_slices* and Bells have *n_slices* [Default 5]. It is even possible to select more than one body. For example
    **-M+s**\ *prism,1/1/1/-5/-10/1*\ **+s**\ *sphere,1/-5* computes the effect of a prism and a sphere. Unfortunately there is
    no current way of selecting distinct densities or magnetic parameters for each body.

>       - *bell,height/sx/sy/z0[/x0/y0/n_sig/npts/n_slices]* Gaussian of height *height* with caracteristic STDs *sx* and *sy*. The base width (at depth *z0*) is controled by the number of sigmas (*n_sig*) [Default = 2]\n");

Reviewer: ```suggestion
      - *bell,height/sx/sy/z0[/x0/y0/n_sig/npts/n_slices]* Gaussian of height *height* with characteristic STDs *sx* and *sy*. The base width (at depth *z0*) is controled by the number of sigmas (*n_sig*) [Default = 2]\n");
```

Examples
--------

> To compute the magnetic anomaly of a cube of unity side located at 5 meters depth and centered at -10,1 in a domain

Reviewer: ```suggestion
To compute the magnetic anomaly of a cube of unit sides located at 5 meters depth and centered at -10,1 in a domain
```
